source/client-side-encryption/client-side-encryption.rst

```diff
@@ -1045,6 +1045,13 @@ ClientEncryption
       getKeyByAltName(keyAltName: String): Optional<Document>;
 
       // Encrypts a BsonValue with a given key and algorithm.
+      // If `opts.queryType` == "rangePreview", `value` is expected to be a BSON document
+      // of one of the following forms:
+      // 1. A Match Expression of this form:
+      //   {$and: [{<field>: {$gt: <value1>}}, {<field>: {$lt: <value2> }}]}
+      // 2. An Aggregate Expression of this form:
+      //   {$and: [{$gt: [<fieldpath>, <value1>]}, {$lt: [<fieldpath>, <value2>]}]
+      // $gt may also be $gte. $lt may also be $lte.
       // Returns an encrypted value (BSON binary of subtype 6). The underlying implementation MAY return an error for prohibited BSON values.
       encrypt(value: BsonValue, opts: EncryptOpts): Binary;
```

Comment:
When trying to implement in Go, I realized the proposed new behavior is backwards breaking. `encrypt` returns type `Binary`. It cannot be changed to return a `Document` when encrypting a Match or Aggregate Expression.

The C driver works around this by returning the output in a [bson_value_t](http://mongoc.org/libmongoc/current/mongoc_client_encryption_encrypt.html)

But other drivers may need a different solution:
- The Go driver [returns a primitive.Binary](https://github.com/mongodb/mongo-go-driver/blob/69630e02f79e0372bbcca779c6d9a73cb27bc915/mongo/client_encryption.go#L119-L121).
- The Java driver [returns a BsonBinary](https://github.com/mongodb/mongo-java-driver/blob/3387038056635e37001af4d551dbc9a66de38513/driver-sync/src/main/com/mongodb/client/vault/ClientEncryption.java#L68-L78).

This was incorrect in the scope. I think this will need to be revised in the scope (possibly a new `encryptQuery` helper?). Regardless, this was my mistake. Please move this PR to draft. I will update this PR after the scope is updated and approved.

Reply:
Added a new `encryptExpression` helper.

Reply:
Has been resolved with a new `encryptExpression` helper.